src/app/cluster/cluster-details/constraints/constraint-dialog/template.html

```
@@ -0,0 +1,52 @@
+<!--
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+<km-dialog-title>{{data.title}}</km-dialog-title>
+<mat-dialog-content>
+  <form [formGroup]="form"
+        fxLayout="column">
+    <mat-form-field fxFlex>
+      <mat-label>Template Name</mat-label>
+      <input required
+             matInput
+             [formControlName]="Controls.Name"
+             type="text"
+             autocomplete="off"
+             cdkFocusInitial>
+      <mat-error *ngIf="form.get(Controls.Name).hasError('required')">
+        Template Name is <strong>required</strong>.
+      </mat-error>
+    </mat-form-field>
+
+    <mat-form-field fxFlex>
+      <mat-label>Constraint Template</mat-label>
+      <mat-select [formControlName]="Controls.ConstraintTemplate"
+                  disableOptionCentering>
+        <mat-option *ngFor="let constraintTemplate of constraintTemplates"
+                    [value]="constraintTemplate.spec.crd.spec.names.kind">
```

I'd add here `?` to be sure that it won't throw undefined error.